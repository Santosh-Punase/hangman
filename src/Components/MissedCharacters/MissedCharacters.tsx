import React from 'react';

import './missed-chars.scss';

interface Props {
  characters: string[];
}

const MissedCharacters: React.FC<Props> = ({ characters }) => {
  if(!characters.length) {
    return null;
  }

  return (
    <div className="hangman__missed-characters">
      <div className="hangman__missed-characters__text">You Missed:</div>
      <div className="hangman__missed-characters__char">
        {characters.map((c) => <div key={c}>{c}</div>)}
      </div>

    </div>
  );
}

export { MissedCharacters };
