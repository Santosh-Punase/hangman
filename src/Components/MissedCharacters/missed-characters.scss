@import 'src/assets/scss/variables';

.hangman__missed-characters {
  display: flex;
  flex-direction: column;
  text-transform: uppercase;
  margin: 2rem 2rem auto auto;
  width: 29rem;
  max-width: 50vw;

  &__text {
    font-size: 1rem;
    margin-left: 0.5rem;
    color: $gray-dark;
  }

  &__char {
    font-size: 3rem;
    font-weight: 800;
    color: $blue-light;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;

    div {
      margin-left: 0.5rem;
    }
  }
}
