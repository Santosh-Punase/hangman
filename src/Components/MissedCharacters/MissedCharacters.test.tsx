import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { MissedCharacters } from './MissedCharacters';

afterEach(cleanup);

describe('Missed characters', () => {

  test('should not render initially', () => {
    const { asFragment } = render(<MissedCharacters characters={[]} />);
    
    expect(asFragment()).toMatchInlineSnapshot(`<DocumentFragment />`);
  });

  test('should render characters', () => {
    const { asFragment } = render(<MissedCharacters characters={['a','b']} />);
    
    expect(asFragment()).toMatchSnapshot();
  });  
});
