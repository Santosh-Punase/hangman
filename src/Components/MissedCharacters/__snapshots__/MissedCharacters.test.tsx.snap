// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missed characters should render characters 1`] = `
<DocumentFragment>
  <div
    class="hangman__missed-characters"
  >
    <div
      class="hangman__missed-characters__text"
    >
      You Missed:
    </div>
    <div
      class="hangman__missed-characters__char"
    >
      <div>
        a
      </div>
      <div>
        b
      </div>
    </div>
  </div>
</DocumentFragment>
`;
